docs/network-requirements.md

```diff
@@ -2,76 +2,44 @@
-    <region-code>.his.arc.azure.com
-    <region>.api.azureml.ms
-    <region>.dp.kubernetesconfiguration.azure.com
-    <region>.experiments.azureml.net
-    acs-mirror.azureedge.net
-    apt.dockerproject.org
-    archive.ubuntu.com
-    changelogs.ubuntu.com
-    cloudflare.docker.com
-    data.policy.core.windows.net
-    dc.services.visualstudio.com
-    gbl.his.arc.azure.com
-    go.microsoft.com
-    guestnotificationservice.azure.com
-    login.microsoftonline.com
-    management.azure.com
-    management.azure.com
-    nvidia.github.io
-    onegetcdn.azureedge.net
-    packages.microsoft.com
-    ppa.launchpad.net
-    pypi.org
-    security.ubuntu.com
-    store.policy.core.windows.net
-    sts.windows.net
-    us.download.nvidia.com
-    ```
+### Azure arc
+
+| Destination Endpoint| Port | Use |
+|--|--|--|
+| management.azure.com | https:443 | Required for the agent to connect to Azure and register the cluster. |
+| <region>.dp.kubernetesconfiguration.azure.com | https:443 | Data plane endpoint for the agent to push status and fetch configuration information. |
+| login.microsoftonline.com | https:443 | Required to fetch and update Azure Resource Manager tokens. |
+|  mcr.microsoft.com| https:443 | Required to pull container images for Azure Arc agents. |
+|  *.data.mcr.microsoft.com| https:443 | Required for MCR storage backed by the Azure content delivery network (CDN). |
+|  gbl.his.arc.azure.com| https:443 | Required to get the regional endpoint for pulling system-assigned Managed Service Identity (MSI) certificates. |
+|  <region-code>.his.arc.azure.com| https:443 | Required to pull system-assigned Managed Service Identity (MSI) certificates. |
+| guestnotificationservice.azure.com, sts.windows.net, *.servicebus.windows.net| https:443 | For Cluster Connect and for Custom Location based scenarios. |
+
+### AML extension
+
+| Destination Endpoint| Port | Use |
+|--|--|--|
+| quay.io, *.quay.io | https:443 | Quay.io registry |
+| gcr.io| https:443 | Google cloud repository |
+| storage.googleapis.com | https:443 | Google cloud storage, gcr images are hosted on |
+| registry-1.docker.io, production.cloudflare.docker.com  | https:443 | Docker hub registry |
+| auth.docker.io| https:443 | Docker repository authentication|
+
+### Inference job
+
+| Destination Endpoint| Port | Use |
+|--|--|--|
+| *.azurecr.io | https:443 | Azure container registry, required for deploying inference and training job|
+| *.blob.core.windows.net | https:443 | Azure blob storage, ACR images are hosted on. required pulling ACR images and downloading model from blob storage|
+| *.workspace.<region>.api.azureml.ms ,  <region>.experiments.azureml.net,  <region>.api.azureml.ms | https:443 | Azure mahince learning service api, required getting credentials of blob storage and ACR|
+| *.kusto.windows.net, *.table.core.windows.net, *.queue.core.windows.net | https:443 | Kusto logs, required for collecting infra components logs |
+
+### Training job
```

Comment:
For training and inference used URLs, If they are needed by both training and inference, can we put the URL to a separate section called ### ML workload. And for which used by training or inference separately, we can put them in current training and inference category.

Reply:
added AML workload